lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -554,12 +554,33 @@ export abstract class ClientApplication {
         try {
             this.preflightBrowserEnvironmentCheck(InteractionType.Silent);
             this.emitEvent(EventType.LOGOUT_START, InteractionType.Redirect, logoutRequest);
-            const validLogoutRequest = this.initializeLogoutRequest(logoutRequest);
-            const authClient = await this.createAuthCodeClient(null, validLogoutRequest && validLogoutRequest.authority);
-            // create logout string and navigate user window to logout. Auth module will clear cache.
-            const logoutUri: string = authClient.getLogoutUri(validLogoutRequest);
-            this.emitEvent(EventType.LOGOUT_SUCCESS, InteractionType.Redirect, validLogoutRequest);
-            return BrowserUtils.navigateWindow(logoutUri, this.config.system.redirectNavigationTimeout, this.logger);
+
+            /*
+             * Before navigating, clear browser cache.
+             * TODO: use sid and iss from query parameter from logout broadcast to clear specific account
+             */
+            if (logoutRequest && logoutRequest.account) {
+                // Clear given account.
+                this.browserStorage.removeAccount(AccountEntity.generateAccountCacheKey(logoutRequest.account));
+            } else {
+                // Clear all accounts and tokens
+                this.browserStorage.clear();
+            }
+
+            if (!BrowserUtils.isInIframe()) {
```

Comment:
There was also an ask in #2127 to only clear cache, regardless of iframe or not. Can we provide a flag on the `EndSessionRequest` object to skip the redirect or do something similar to `onRedirectNavigate`?

Reply:
Yeah, we can do that.

Reply:
The `onRedirectNavigate` PR will add this for logout.